be/src/storage/schema_change_utils.cpp

```diff
@@ -538,16 +552,26 @@ Status SchemaChangeUtils::parse_request(const TabletSchema& base_schema, const T
         ColumnMapping* column_mapping = chunk_changer->get_mutable_column_mapping(i);
 
         if (materialized_view_param_map.find(column_name) != materialized_view_param_map.end()) {
-            AlterMaterializedViewParam mvParam = materialized_view_param_map.find(column_name)->second;
-            column_mapping->materialized_function = mvParam.mv_expr;
+            auto& mvParam = materialized_view_param_map.find(column_name)->second;
+            if (mvParam.mv_expr != nullptr) {
+                chunk_changer->set_has_mv_expr_context(true);
+                chunk_changer->init_runtime_state(TQueryOptions(), TQueryGlobals());
```

Comment:
TQueryOptions and TQueryGlobals maybe need inited by FE, otherwise some function like Time relative will not be correct, maybe should test for this function just in case

Reply:
fix it.